### PR TITLE
refactor: refine logging configuration

### LINKED ISSUE #6034: bug: risingwave logs should use localtime.
### Describe the bug

Risingwave's log is using some fixed time zone, but it should use localtime to improve user experience.

### To Reproduce

image
image

### Expected behavior

Report log using local time.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Fallback to default levels for most of the RisingWave crates.
- Introduce back local time for logging. See [comment here](https://github.com/risingwavelabs/risingwave/pull/6152#discussion_r1234783018). Close #6034.
- Simplify env-var overrides without relying on `EnvFilter`.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/utils/runtime/src/lib.rs [hunk 1]
```diff
@@ -144,8 +138,21 @@ pub fn set_panic_hook() {
...
+    let default_timer = {
+        let timer = OffsetTime::local_rfc_3339().unwrap_or_else(|e| {
+            println!("failed to get local time offset: {e}, falling back to UTC");
+            OffsetTime::new(
+                time::UtcOffset::UTC,
+                time::format_description::well_known::Rfc3339,
+            )
+        });
+
+        move || timer.clone()
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:151]: Why does this need `clone`? 😲

File: src/utils/runtime/src/lib.rs [hunk 2]
```diff
@@ -165,22 +172,31 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
         let filter = configure_risingwave_targets_fmt(filter);
 
         // For all other crates
-        let filter = filter.with_default(if cfg!(debug_assertions) {
+        let filter = filter.with_default(if cfg!(debug_assertions) && !is_ci() {
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:175]: Should CI have more logs? I don't know 🤣

File: src/utils/runtime/src/lib.rs (hunk 1 above)
Comment (xxchan) [reply src/utils/runtime/src/lib.rs:151]: Oh, it will become `FnOnce`

Comment (xxchan) [reply src/utils/runtime/src/lib.rs:151]: BTW, why not `fn`

**Review by zwang28 — APPROVED** [review]
LGTM

File: src/utils/runtime/src/lib.rs (hunk 1 above)
Comment (xxchan) [reply src/utils/runtime/src/lib.rs:151]: Wait, why not `let timer` + `clone` later directly

File: src/utils/runtime/src/lib.rs [hunk 3]
```diff
@@ -144,8 +138,21 @@ pub fn set_panic_hook() {
...
+                time::format_description::well_known::Rfc3339,
+            )
+        });
+
+        move || timer.clone()
+    };
+
     // Default filter for logging to stdout and tracing.
-    let filter = {
+    let default_filter = {
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:155]: ditto, I don't understand why this needs to be a closure

File: src/utils/runtime/src/lib.rs [hunk 4]
```diff
@@ -165,22 +172,31 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
...
-            });
+        // Overrides from settings
+        let filter = filter.with_targets(settings.targets);
 
-        move |additional_targets: Vec<(&str, Level)>| {
-            to_env_filter(filter.clone().with_targets(additional_targets))
-        }
+        // Overrides from env var
+        let filter = if let Ok(rust_log) = std::env::var(EnvFilter::DEFAULT_ENV) && !rust_log.is_empty() {
+            let rust_log_targets: Targets = rust_log.parse().unwrap();
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:186]:
```suggestion
            let rust_log_targets: Targets = rust_log.parse().expect("failed to parse RUST_LOG");
```

File: src/utils/runtime/src/lib.rs [hunk 5]
```diff
@@ -165,22 +172,31 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
...
+            if let Some(default_level) = rust_log_targets.default_level() {
+                filter
+                    .with_targets(rust_log_targets)
+                    .with_default(default_level)
+            } else {
+                filter.with_targets(rust_log_targets)
+            }
+        } else {
+            filter
+        };
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:197]: a little noodle 🤣

File: src/utils/runtime/src/lib.rs [hunk 6]
```diff
@@ -199,9 +216,7 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
         layers.push(
             fmt_layer
                 .with_filter(FilterFn::new(|metadata| metadata.is_event())) // filter-out all span-related info
-                .with_filter(filter(vec![
-                    ("rw_tracing", Level::OFF), // filter out tracing-only events
-                ]))
+                .with_filter(default_filter().with_target("rw_tracing", Level::OFF)) // filter-out tracing-only events
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:219]: Would it be better to opt-in rw_tracing in tracing layer instead of opting-out here?

Comment (BugenZhao) [reply src/utils/runtime/src/lib.rs:219]: Do you mean setting `rw_tracing` to `OFF` in `default_filter`?

Comment (xxchan) [reply src/utils/runtime/src/lib.rs:219]: yes

Comment (BugenZhao) [reply src/utils/runtime/src/lib.rs]: I don't find a way to "cancel" an `OFF` elegantly, or we have to retrieve the default level and set it again. 😕

File: src/utils/runtime/src/lib.rs (hunk 3 above)
Comment (BugenZhao) [reply src/utils/runtime/src/lib.rs:151]: Previously we use `EnvFilter` here, which does not impl `Clone`. Some conventional thinking. 🥰

**Review by xxchan — APPROVED** [review]
happy

**Pull request merged**